xarray/backends/api.py

```diff
@@ -503,7 +498,7 @@ def open_dataset(
         **decoders,
         **kwargs,
     )
-    ds = _dataset_from_backend_dataset(
```

Comment:
This is an example where I think the intermediate local variable created by a multi-line statement improves readability as having `return` on the last line of the function.

Reply:
> I feel this PR is too big and diverse to accept it as a whole.
> 
> For example I like most changes from older style formatting to f-strings, but I disagree strongly the moving the `return` statements into functions improves readability.

While I don't have the same sense on the specifics, if people feel strongly then it may be reasonable to revert or pause on some of these.

What do you suggest re the broader PR though? While focused PRs are easier to review and reach agreement — now this is here, it does seem on net beneficial, and we should take advantage of the benefit.

Are there a few items you feel strongly about that we could revert and then merge the rest?